samcli/local/init/__init__.py

```diff
@@ -13,29 +13,53 @@
-    "python": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
-    "ruby2.5": os.path.join(_templates, "cookiecutter-aws-sam-hello-ruby"),
-    "nodejs6.10": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs6"),
-    "nodejs8.10": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs"),
-    "nodejs": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs"),
-    "dotnetcore2.0": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "dotnetcore2.1": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "dotnetcore1.0": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "dotnetcore": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "dotnet": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
-    "go1.x": os.path.join(_templates, "cookiecutter-aws-sam-hello-golang"),
-    "go": os.path.join(_templates, "cookiecutter-aws-sam-hello-golang"),
-    "java8": os.path.join(_templates, "cookiecutter-aws-sam-hello-java"),
-    "java": os.path.join(_templates, "cookiecutter-aws-sam-hello-java")
+RUNTIME_DEP_TEMPLATE_MAPPING = {
+    "python3.7_pip": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
+    "python3.6_pip": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
+    "python2.7_pip": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
+    "python_pip": os.path.join(_templates, "cookiecutter-aws-sam-hello-python"),
+    "ruby_bundler": os.path.join(_templates, "cookiecutter-aws-sam-hello-ruby"),
+    "ruby2.5_bundler": os.path.join(_templates, "cookiecutter-aws-sam-hello-ruby"),
+    "nodejs6.10_npm": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs6"),
+    "nodejs8.10_npm": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs"),
+    "nodejs_npm": os.path.join(_templates, "cookiecutter-aws-sam-hello-nodejs"),
+    "dotnetcore2.0_cli-package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "dotnetcore2.1_cli-package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "dotnetcore1.0_cli-package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "dotnetcore_cli_package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "dotnet_cli-package": os.path.join(_templates, "cookiecutter-aws-sam-hello-dotnet"),
+    "go1.x_go-get": os.path.join(_templates, "cookiecutter-aws-sam-hello-golang"),
+    "go_go-get": os.path.join(_templates, "cookiecutter-aws-sam-hello-golang"),
+    "java8_maven": os.path.join(_templates, "cookiecutter-aws-sam-hello-java-maven"),
+    "java_maven": os.path.join(_templates, "cookiecutter-aws-sam-hello-java-maven"),
+    "java8_gradle": os.path.join(_templates, "cookiecutter-aws-sam-hello-java-gradle"),
+    "java_gradle": os.path.join(_templates, "cookiecutter-aws-sam-hello-java-gradle")
+}
+
+RUNTIME_DEP_MAPPING = {
```

Comment:
Why not encapsulate this into the [Runtimes enum](https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/local/docker/lambda_image.py#L23) to keep all of this into one place?

Reply:
Maybe, I need a double mapping though. Language -> Dep manager and language, dep combo to template. will think about it.